packages/design-system/src/components/SearchBar/style.scss

```diff
@@ -0,0 +1,70 @@
+@import '../../scss/variables/colors';
+
+$base-class: "search-bar";
+
+// TODO: should container have fixed width and height?
+.#{$base-class}__container {
+  display: inline-flex;
+  position: relative;
+  flex-direction: column;
+  justify-content: center;
+  height: 32px;
+  width: 290px;
+}
+
+.#{$base-class}__icon {
+  position: absolute;
+  align-self: center;
+
+  &--search {
+    @extend .#{$base-class}__icon;
+    left: 0;
+    padding: 7px 13px;
+
+    &-compact {
+      cursor: pointer;
+    }
+  }
+
+  &--close {
+    @extend .#{$base-class}__icon;
+    right: 13px;
+    cursor: pointer;
+  }
+
+  &--loader {
+    @extend .#{$base-class}__icon;
+    right: 13px;
+  }
+
+}
+
+.#{$base-class}__input {
+  width: 100%;
+  height: 100%;
+  box-sizing: border-box;
+  padding: 0 40px;
+  border-radius: 4px;
+  border: 1px solid #cfd3d5;
+  transition: width ease-in 0.3s;
+
+  &:focus, &:active {
+    outline: none;
+    border-color: $primary-color
+  }
+}
+
+.#{$base-class}__input-compact {
+  width: 0;
+  padding-right: 0;
```

Comment:
the animation is jumping because of this padding-right reset. Can we improve that?

Reply:
It's improved :) I added width and padding to animation